### PR TITLE
Strand-filter and strand-pagination components. 

### PR DESCRIPTION
**Pull request by passionintellectual**
Hi,
I have added few components for like strand-injector, strand-data-provider (for any processing of data and UI seperation), strand-pagination, strand-filter.

Please let me know in case, I missed any step or any code-changes or architectural changes required. 
I will quickly implement those.

Thanks and Regards,

Ganesh Nemade.

### CODE REVIEW

**Comment by dlasky** [comment]
A large amount of code exists for the majority of this.

see mm-sync, mm-ajax, request.html and the ajax plugin system which handles pagination

File: src/strand-filter/doc.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+	"name":"mm-filter",
```
Comment (dlasky) [inline src/strand-filter/doc.json]: check out strand-array-munge

**Comment by passionintellectual** [comment]
@dlasky I am taking this oportunity to thank MediaMath for offering such a scalable library to community. I enjoyed using these controls. 

mm-sync, mm-ajax and ajax plugin is dealing with serverside communication. The components I came up would be more of data-source agnostic. I believe, this would help users experience the strand controls in the same way irrespective of where the data is coming, (e.g. client side, localstorage, server or anything like pouchdb/indexdb etc) 

Strand-pagination and strand-filter are only providing the UI making no calls to any data source. But at the same time, it integrates with Strand-data-component which handles data processing like actual searching ( in case of strand-filter) . We can extend strand-data-component to communicate with mm-ajax and mm-sync internally with more configurations.

However, I may be missing something or be on the wrong track. Please guide in this case. I would like to know your views. 
And thanks for your time.

**Comment by dlasky** [comment]
I think thats partially a consequence of pushing too much stuff in a single PR, we try to separate them by a single component, or in extreme cases a tightly coupled set of components that are generally parent > child of eachother.

In terms of UX we don't want to allow for pagination.  You'll notice that `strand-item-recycler` has an index property--which is sufficient to determine the page for pagable scenarios in dropdowns and grids.  When sync is properly extended the indicies of the data source and the desired target can be bound and new pages are called and spliced into the source array on the fly.

The search stuff in particular is entirely to opinionated about data and is better left to the client application.

**Comment by dlasky** [comment]
closing as stale

**Pull request closed without merging**